index.js

```diff
@@ -182,18 +183,29 @@ module.exports = fp(
             metricsConfig.routes.errors ||
             metricsConfig.routes.hits
         ) {
+            fastify.addHook('onRoute', (options) => {
+                // TODO: check for duplicates when registering routes
+                options.config = options.config || {};
+                options.config.metrics = options.config.metrics || {};
+                options.config.metrics.fastifyPrefix = normalizeFastifyPrefix(
+                    options.prefix
+                );
+                options.config.metrics.routesPrefix =
+                    normalizeRoutePrefix(prefix);
+            });
             if (mode === 'dynamic') {
                 let getLabel =
                     metricsConfig.routes.getLabel || dynamicMode.getLabel;
                 if (typeof getLabel !== 'function') {
                     throw new Error('"getLabel" must be a function.');
                 }
-                fastify.decorateRequest('metricsLabel', '');
-                fastify.decorateReply('metricsLabel', '');
+                fastify.decorateRequest('_metricsLabel', '');
+                fastify.decorateReply('_metricsLabel', '');
```

Comment:
I would like to try a Symbol here, I am no sure what will happen with the encapsulation.

Reply:
After some tests, it looks like it doesn't break anything. I'll try it.

Reply:
Done. The reason behind this is that I want to expose a unified way to access the label (#84 ) and hide the internal field used by the plugin.